src/App.js

```diff
@@ -1,10 +1,76 @@
+import { useState, useEffect, useCallback } from "react";
+import TodoList from "./TodoList";
+import TodoInputForm from "./TodoInputForm";
+import styled from "styled-components";
+
+const Container = styled.div`
+  height: 100vh;
+  display: flex;
+  justify-content: center;
+  align-items: center;
+`;
+const Box = styled.main`
+  width: 350px;
+  height: 600px;
+  background-color: white;
+  border-radius: 20px;
+  box-shadow: 1px 1px 30px grey;
+`;
+
+const App = () => {
+  const [todoList, setTodoList] = useState([]);
+  const [doneList, setDoneList] = useState([]);
+
+  // save to localStorage
+  const _saveLocalStorage = useCallback((type, list) => {
+    localStorage.setItem(type, JSON.stringify(list));
+  }, []);
+
+  // 렌더링 시 localStorage에서 todo list 불러오기
+  useEffect(() => {
+    const todoStorage = localStorage.getItem("todoList");
+    const doneStorage = localStorage.getItem("doneList");
+
+    // localStorage에 저장된 list가 있는지 확인
+    if (todoStorage) {
+      const loadTodo = JSON.parse(todoStorage);
+      setTodoList(loadTodo);
+    }
+    if (doneStorage) {
+      const loadDone = JSON.parse(doneStorage);
+      setDoneList(loadDone);
+    }
+  }, []);
```

Comment:
```suggestion

  const todoStorage = localStorage.getItem("todoList");
  const doneStorage = localStorage.getItem("doneList");

  // localStorage에 저장된 list가 있는지 확인
  if (todoStorage) {
    const loadTodo = JSON.parse(todoStorage);
  }
  if (doneStorage) {
    const loadDone = JSON.parse(doneStorage);
  }

const App = () => {
  const [todoList, setTodoList] = useState([loadTodo]);
  const [doneList, setDoneList] = useState([loadDone]);

  // save to localStorage
  const _saveLocalStorage = useCallback((type, list) => {
    localStorage.setItem(type, JSON.stringify(list));
  }, []);
```

[]을  dep으로 두면 매 렌더링마다 localStorage를 긁어오는 것 같아
이런 식으로 외부로 빼고 initial state로 두는 건 어떨지 코멘트 드려봅니다.

Reply:
오 그렇네요... 외부로 빼는거 좋은 방법인 것 같아요! 감사합니다~!